main.cpp

```diff
@@ -70,29 +70,440 @@ write 3 UDTs below that EACH have:
  You will use those in Part 3 of this project.
 
  */
+#include <iostream>
 
 /*
  copied UDT 1:
  */
 
+struct City
+{
+    City();
+    ~City();
+
+    std::string name, country;
+    std::string newLawName = "Friday's Off";
+    int population = 5000000;
+    float latitude = 43.6532f;    
+    float longitude = -79.3470f;  
+
+    void expand(float expansionRate = 1.1f);
+    std::string createLaw(); 
+    int updatePopulation(int immigrantsYear, int emigrantsYear, int birthsYear, int deathsYear, int years);  
+
+    struct PoliceDepartment
+    {
+        PoliceDepartment();
+        ~PoliceDepartment();
+
+        int staff, budgetMillions, salary;
+        float crimeRate = 0.f;
+        std::string chief = "unnamedPoliceChief";
+
+        float getConvictionRate(float arrests, float convictions);
+        void trainRookies(int rookies, int monthsTraining);
+        int checkStaffCost(int numberStaff, int monthlySalary);
+    };
+};
+
+City::City() : name("Montreal"), country("Canada"), population(5000000)
+{
+    std::cout << "City being constructed." << std::endl;
+}
+
+City::~City()
+{
+    std::cout << "City being deconstructed." << std::endl;
+}
+
+City::PoliceDepartment::PoliceDepartment() : staff(4500), budgetMillions(500), salary(85000)
+{
+    std::cout << "PoliceDepartment being constructed." << std::endl;
+}
+
+City::PoliceDepartment::~PoliceDepartment() 
+{
+    std::cout << "PoliceDepartment being deconstructed." << std::endl;
+}
+
+
+void City::expand(float expansionRate)
+{
+    if(expansionRate < 0.f)
+    {
+        std::cout << "we're shrinking!" << std::endl;    
+    } 
+}
+
+std::string City::createLaw()
+{
+  return newLawName;  
+}
+
+int City::updatePopulation(int immigrantsYear, int emigrantsYear, int birthsYear, int deathsYear, int years)
+{
+    for(int i = 0; i <= years; ++i)
+    {
+        population = population + immigrantsYear - emigrantsYear + birthsYear - deathsYear;
+    }    
+    return population;
+}
+
+float City::PoliceDepartment::getConvictionRate(float arrests, float convictions)
+{
+    std::cout << convictions / arrests << " is the conviction rate." << std::endl;
+    return convictions / arrests;
+}
```

Comment:
D.R.Y. 
you're computing this value twice. 
cache it into a local variable. 
use it in the std::cout statement and the return statement
- [ ] revise